admin/services/pin-generation.service.js

```diff
@@ -90,8 +92,11 @@ pinGenerationService.generatePupilPins = async (pupilsList) => {
 pinGenerationService.generateSchoolPassword = (school) => {
   let { schoolPin, pinExpiresAt } = school
   if (!pinValidator.isActivePin(schoolPin, pinExpiresAt)) {
-    const length = 8
-    school.schoolPin = pinGenerationService.generateRandomPin(length)
+    const allowedWords = (config.Data.allowedWords && config.Data.allowedWords.split(',')) || []
+    const firstRandomWord = allowedWords.length > 0 ? allowedWords[Math.floor(Math.random() * allowedWords.length)] : ''
```

Comment:
Math.random() does not have enough unpredictability to be used for generating passwords.  Please use the crypto library instead.  https://nodejs.org/api/crypto.html

I suspect that reading and splitting the csv word list at startup It would also be better to parse the allowedWord list at startup and then keep it as as a data structure (e..g a Set) during operation.  

There are also limits to size of data that can be kept in environment variables - let's discuss this in private chat.

Reply:
@jon-shipley updated with randomBytes method usage from crypto lib